iota/transaction/creation.py

```diff
@@ -387,6 +505,20 @@ def sign_inputs(self, key_generator):
         # type: (KeyGenerator) -> None
         """
         Sign inputs in a finalized bundle.
+
+        Generates the necessary cryptographic signatures to authorize spending
+        the inputs. You do not need to invoke this method if the bundle does
+        not contain any transactions that spend IOTAs.
+
+        :param KeyGenerator key_generator:
+            Generator to create private keys for signing.
+
+        :raises RuntimeError: if bundle is not yet finalized.
+        :raises ValueError:
+            - if the input transaction doesn't have ``key_index`` attribute
+              defined.
+            - if the input transaction doesn't have ``security_level`` attribute
+              defined.
```

Comment:
❔ Are these attributes of the input transaction, or of the address specified in the input transaction?

E.g., `- if the input transaction specifies an address that doesn't have...`

Reply:
good catch!